dao/src/main/java/greencity/repository/UserTableRepo.java

```diff
@@ -81,11 +86,37 @@ private Predicate getPredicateForWhere(UserFilterCriteria us, Root<User> userRoo
         return criteriaBuilder.and(predicateList.toArray(new Predicate[0]));
     }
 
+    private void searchUsers(UserFilterCriteria us, Root<User> userRoot, List<Predicate> predicateList) {
+        Optional<Join<User, ?>> orderJoin = userRoot.getJoins().stream().findFirst();
+        Predicate predicate = null;
+        if (orderJoin.isPresent()) {
+            Expression<Long> expression = criteriaBuilder.count(orderJoin.get().get("user"));
+            predicate = criteriaBuilder.or(
+                criteriaBuilder.like((userRoot.get(DATE_OF_REGISTRATION)).as(String.class),
+                    "%" + us.getSearch().toUpperCase() + "%"),
```

Comment:
think about extracting all the like type criteria to some Util class since there is duplication every time: "%" + text + "%" etc.

Reply:
Added an additional private method that implements logic